src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -47,11 +47,13 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
         private async Task<(TLocalDeclarationSyntax, ILocalSymbol)> FindDisposableLocalDeclaration(Document document, TextSpan selection, CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
+            var declarationSyntax = await refactoringHelperService.TryGetSelectedNodeAsync<TLocalDeclarationSyntax>(document, selection, cancellationToken).ConfigureAwait(false);
```

Comment:
I would adjust this to look for either a local declaration or initializer of a local variable of disposable type.

Reply:
Wholeheartedly agree but the rest of the refactoring works with declaration and refactoring the inner workings of that refactoring is out of current scope.